connector/protobuf/src/test/scala/org/apache/spark/sql/protobuf/ProtobufFunctionsSuite.scala

```diff
@@ -1124,7 +1124,55 @@ class ProtobufFunctionsSuite extends QueryTest with SharedSparkSession with Prot
     }
   }
 
-  test("Corner case: empty recursive proto fields should be dropped") {
+  test("retain empty proto fields") {
+    val options = Map("recursive.fields.max.depth" -> "4", "retain.empty.message" -> "true")
+
+    // EmptyRecursiveProto at the top level. It will be an empty struct.
+    checkWithFileAndClassName("EmptyProto") {
+      case (name, descFilePathOpt) =>
+        val df = emptyBinaryDF.select(
+          from_protobuf_wrapper($"binary", name, descFilePathOpt, options).as("empty_proto")
+        )
+        // Top level empty message is retained without adding dummy column to the schema.
+        assert(df.schema == structFromDDL("empty_proto struct<>"))
```

Comment:
Why do we have empty struct here even though we have set the option?

Reply:
Because this is top level empty struct, only nested level empty struct is not allowed.

Reply:
Interesting. Is this tested in the parquet test below?

Reply:
Please add this example in the documentation.
Btw, is it better to be consistent with the dummy field?

Reply:
Example added to the documentation, I feel it ok to not add dummy column to top level empty struct because I can't think of any use case for it.

Reply:
Why is that? Isn't it better to be consistent? "If there is an empty protobuf, it will have a dummy column". 
The users don't distinguish between top level vs nested.
Btw, does parquet test below include empty top level struct? Could you point to it?

Reply:
I don't have a strong opinion here, changed to add dummy column also for top level empty strcut.

Reply:
Is this changed merged?

Reply:
Yes, now the top level also get the dummy column if the option is true.

Reply:
LGTM